### DISCUSSION TITLE
What can be done using WOQL that can't be done using GQL?

### DISCUSSION THREAD

**Discussion by annainfo** [terminusdb/terminusdb#2089] [Q&A]
What can be done using WOQL that can't be done using GQL?

**Comment by alexander-mart** [comment]
@annainfo Key and common term is a 'graph'. Graph is mathematical concept? Or not? Why we need graphs to represend our data?

## 1. What do 'graphs' mean in WOQL

- WOQL actually is Datalog — [declarative](https://en.wikipedia.org/wiki/Declarative_programming) [logic programming](https://en.wikipedia.org/wiki/Logic_programming) language:
https://en.wikipedia.org/wiki/Datalog

- WOQL provide declarative logic programming for logically correct answers based on [Linked Data](https://en.wikipedia.org/wiki/Linked_data) inside our database.

- Modern Linked Data use a [JSON-LD](https://en.wikipedia.org/wiki/JSON-LD) serialization, and WOQL-query themself is serialized to JSON-LD too that make queries are [first-class citizens](https://en.wikipedia.org/wiki/First-class_citizen) and allowing them to be construct and stored as structured data.


## 2. What do 'graphs' mean in GraphQL

Also, GraphQL has little relation to Linked Data, although it can be used on top of it

Why GraphQL is Not a Graph Database
https://www.apollographql.com/blog/what-is-a-graph-database-why-graphql-is-not-a-graph-database

TL;DR

> What is the relationship between GraphQL and graph databases?
> Fundamentally, there is no relationship. Besides the fact that GraphQL and graph databases have the word “graph” in their name, they were designed for two entirely separate purposes.
> 
> GraphQL, the query language, was created to make it easier for app developers fetch data by asking specifically for what they want, while graph databases were designed to allow for performant analysis of relationships between entities.

**Reply by alexander-mart** [reply]
UPD: Now I see, you talk about «[GQL](https://github.com/orgs/terminusdb/discussions/2088)», not «GraphQL»

- https://en.wikipedia.org/wiki/Graph_Query_Language
- https://jtc1info.org/slug/gql-database-language/

As we can see, there is currently no implementation of this standard. In addition, the standard says «GQL is a database language, not just a graph query language». This will probably require writing a database core.

**Reply by annainfo** [reply]
I don't see what you claim "we" can see. But the question had no relation to the existence of any GQL implementation anyway.

**Reply by alexander-mart** [reply]
@annainfo To understand what exactly is not in GQL, what is in WOQL, you need to read the documentation. Do you have the opportunity to download the document published in 2025-04?
